isis/src/qisis/objs/StretchTool/StretchTool.cpp

```diff
@@ -370,6 +380,170 @@ namespace Isis {
     }
   }
 
+
+  /**
+   * Restores a saved stretch from the cube
+   */
+  void StretchTool::loadStretchFromCube(){
```

Comment:
All the code that interacts with the `Cube` is in `StretchTool` because the other classes don't have access to the cube via the viewport.